nixos/modules/services/networking/dnsdist.nix

```diff
@@ -35,25 +35,17 @@ in {
     };
   };
 
-  config = mkIf config.services.dnsdist.enable {
-    systemd.services.dnsdist = {
-      description = "dnsdist load balancer";
-      wantedBy = [ "multi-user.target" ];
-      after = ["network.target"];
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.dnsdist ];
 
+    systemd.services.dnsdist = {
       serviceConfig = {
-        Restart="on-failure";
-        RestartSec="1";
+        wantedBy = [ "multi-user.target" ];
```

Comment:
You made a mistake here: this should be in top-level, not `serviceConfig`.
It would be really useful to have nixos call `systemd-verify` or something to avoid this kind of issue.

Reply:
Good catch. I'm left wondering why the service automatically started when I ran it locally, though... :thinking: